tests/ecosystem/upgrade/test_upgrade_sc_allowexpansion_false.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier1,
+    bugzilla,
+    skipif_ocs_version,
+)
+from ocs_ci.framework.pytest_customization.marks import (
+    pre_ocs_upgrade,
+    post_ocs_upgrade,
+)
+from ocs_ci.ocs.resources.pod import (
+    wait_for_storage_pods,
+    get_ocs_operator_pod,
+    get_pod_logs,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@bugzilla("2125815")
+@skipif_ocs_version("<4.11")
```

Comment:
What if we will run Z-stream upgrade from 4.11.5 to 4.11.6 , is this also applicable? As this skip marker will not prevent this but in upgrade steps you describe upgrade to 4.12.

Reply:
Its not applicable for 4.11 , Because the fix was planned only for 4.12.
https://bugzilla.redhat.com/show_bug.cgi?id=2125815#c19

Reply:
Then we will need to update the logic here and make it more robust to also check upgrade version is 4.12. And if OCS version is 4.11, and upgrade version 4.12, run the test, otherwise skip the test.